src/docfx/config/Config.cs

```diff
@@ -210,11 +211,20 @@ private static (List<Error>, Config) LoadCore(string docsetPath, string configPa
                 errors.AddRange(extendErrors);
             }
 
+            finalConfigObject = OverwriteConfig(finalConfigObject, options.Locale, GetBranch());
+
             var deserializeErrors = new List<Error>();
             (deserializeErrors, config) = JsonUtility.ToObject<Config>(finalConfigObject);
             errors.AddRange(deserializeErrors);
 
             return (errors, config);
+
+            string GetBranch()
```

Comment:
nit: can we call `GetBranch` only when there is branch override ?

Reply:
that's actually not cost too much, we only load config 1 or 2 times, comparing the code complex to implement the OnDemondGetBranch